### PR TITLE
Add CSPM Severity Framework to Docs Site

### PR DESCRIPTION
**Pull request by Frichetten**
### What does this PR do? What is the motivation?
This PR adds the new CSPM severity framework to the public documentation site, along with some examples of how to use it. 

### Merge instructions
Please do **not** merge this immediately. It needs to be reviewed by some other parties first.

### Additional notes
This work is tied to [this](https://datadoghq.atlassian.net/browse/SECRES-1422?atlOrigin=eyJpIjoiOWJkMTJjM2U3YzdhNGM3MjkxZTVjYjNkYWYzNWY2MTciLCJwIjoiaiJ9) ticket.

### CODE REVIEW

File: content/en/security/severity_scoring.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+|:--------:|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+|    Low   | This misconfiguration is related to security hardening, hygiene, resource metadata, or industry best practice configurations. By itself, this misconfiguration represents little to no impact to the environment.                                                                                                             |
+|  Medium  | Abusing this misconfiguration results in an impact to the confidentiality, integrity, or availability of the vulnerable component or its directly associated resources.                                                                   |
+|   High   | Abusing this misconfiguration results in an impact to the following: confidentiality, integrity or availability of the vulnerable component and impacts a significant number of other resources (E.G. S3FullAccess, EC2FullAccess, etc.). |
```
Comment (christophetd) [inline content/en/security/severity_scoring.md:50]: the sentence makes it sound like S3FullAccess is "another resource", would rephrase slightly and use inline code blocks for the policy names

**Review by christophetd — COMMENTED** [review]
Looks great overall, left a few comments!

File: content/en/security/severity_scoring.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+---
+title: Severity Scoring
+kind: documentation
+further_reading:
+- link: "/security/misconfigurations/"
```
[the remark this answers is not in the document]
Comment (Frichetten) [reply content/en/security/severity_scoring.md:5]: Thats a good call. I've added identity risks and CSM Vulnerabilities.

File: content/en/security/severity_scoring.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+The likelihood component is made up of two subcomponents; The attack vector, the means through which a misconfiguration can be exploited, and the accessibility, if the resource is publicly accessible or not.
+
+|               |                     | Accessibility |          |
+|---------------|---------------------|---------------|----------|
+|               |                     | **Private**       | **Public**   |
+| **Attack Vector** | **Required Privileges** | Low           | Medium   |
+|               | **Vulnerability**       | Medium        | High     |
+|               | **No Authorization**    | High          | Critical |
+
+The attack vector is determined by the following criteria:
```
[the remark this answers is not in the document]
Comment (Frichetten) [reply content/en/security/severity_scoring.md:30]: Hey sorry, not sure what you mean? `likelihood` has a table as well.

File: content/en/security/severity_scoring.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  text: "Learn more about CSM Misconfigurations"
+---
+
+## Cloud Security Management
+
+### Misconfigurations and Identity Risks
+
+Starting in early 2024, all CSM Misconfigurations and Identity Risk rules will migrate to our severity score framework. This framework is designed to compare the likelihood that an adversary will take advantage of a misconfiguration to the risk posed to your environment. By weighting both of these aspects, findings can be prioritized more accurately by their real-world risks. The matrices below show how to compute a misconfiguration's severity score depending on certain criteria.
+
+#### Likelihood
```
[the remark this answers is not in the document]
Comment (christophetd) [reply content/en/security/severity_scoring.md:16]: reat!

File: content/en/security/severity_scoring.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+- link: "/security/misconfigurations/"
+  tag: "Documentation"
+  text: "Learn more about CSM Misconfigurations"
+---
+
+## Cloud Security Management
+
+### Misconfigurations and Identity Risks
+
+Starting in early 2024, all CSM Misconfigurations and Identity Risk rules will migrate to our severity score framework. This framework is designed to compare the likelihood that an adversary will take advantage of a misconfiguration to the risk posed to your environment. By weighting both of these aspects, findings can be prioritized more accurately by their real-world risks. The matrices below show how to compute a misconfiguration's severity score depending on certain criteria.
```
[the remark this answers is not in the document]
Comment (Frichetten) [reply content/en/security/severity_scoring.md]: Ah, that was actually a request from product. They want the docs to be available before we start migrating rules.

[the remark this answers is not in the document]
Comment (christophetd) [reply content/en/security/severity_scoring.md]: Who? I think we might want to challenge that. I feel like this gives a bad look, "we're showing that we have no clue how to assign a severity, and doing nothing about it"

File: content/en/security/severity_scoring.md (hunk 1 above)
Comment (Frichetten) [reply content/en/security/severity_scoring.md]: Changed the language and added the `code block` where needed.

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**